src/map/vending.cpp

```diff
@@ -204,7 +204,7 @@ void vending_purchasereq(struct map_session_data* sd, int aid, int uid, const ui
 		case CHKADDITEM_EXIST:
 			break;	//We'd add this item to the existing one (in buyers inventory)
 		case CHKADDITEM_NEW:
-			new_++;
+			new_ += itemdb_inventory_slot_needed(itemdb_search(vsd->cart.u.items_cart[idx].nameid), amount);
```

Comment:
shouldn't need here, if the vended item is GUID'd item with a stack of 10, player should get it as 10 in 1 inventory slot
```suggestion
			new_++;
```